Makefile

```diff
@@ -191,16 +229,19 @@ calico-build/centos6:
 	  docker build \
 	  --build-arg=UID=$(MY_UID) \
 	  --build-arg=GID=$(MY_GID) \
-	  -f centos6-build.Dockerfile$(ARCHTAG) \
+	  -f centos6-build.Dockerfile.$(ARCH) \
 	  -t calico-build/centos6 .
 
 # Build the calico/felix docker image, which contains only Felix.
-.PHONY: calico/felix
-calico/felix: bin/calico-felix
+.PHONY: calico/felix calico/felix-$(ARCH) register
+
+# by default, build the image for the target architecture
+calico/felix: calico/felix-$(ARCH)
+calico/felix-$(ARCH): bin/calico-felix-$(ARCH) register
 	rm -rf docker-image/bin
 	mkdir -p docker-image/bin
-	cp bin/calico-felix docker-image/bin/
-	docker build --pull -t $(FELIX_IMAGE_NAME) --file ./docker-image/Dockerfile$(ARCHTAG) docker-image
+	cp bin/calico-felix-$(ARCH) docker-image/bin/
+	docker build --pull -t calico/felix:latest-$(ARCH) --file ./docker-image/Dockerfile.$(ARCH) docker-image
```

Comment:
I read this comment through several times, still don't get it. We cannot run this as `docker build --pull -t calico/felix:latest ... ` (as opposed to `calico/felix:latest-$(ARCH)` ), since it is building an arch-specific image. If we do, we can trounce other images.

Or are you saying that in _addition_ to the `ARCH`-specific tag to provide a `calico/felix:latest` tag if `ARCH=amd64`? We do that on [LL 638-643](https://github.com/projectcalico/felix/pull/1703/files#diff-b67911656ef5d18c4ae36cb6741b7965R638)

Reply:
I mean "in addition".  Lines 638-643 are in the wrong target for our CI systems.

For context, our SemaphoreCI script looks like a bit like this:
```
make calico/felix
docker tag calico/felix:latest calico/felix:`git describe`
docker push  calico/felix:`git describe`
```

If you do the retagging in the release-once-tagged target then I think the Semaphore script will break.  I'd like to keep the old names for now, until we've got the whole pipeline ready for multi-arch, then we only need to update Semaphore once.

Updating Semaphore is a bit painful because ideally, we need it to keep building old release branches too.  That means that every change needs to be put behind version detection logic, which is a pain.

I repushed your branch to the main repo so that the above logic will run (it's normally blocked for PRs) and it does fail: https://semaphoreci.com/calico/felix-2/branches/smc-multi-arch/builds/1

Reply:
> I mean "in addition". Lines 638-643 are in the wrong target for our CI systems.

Ah, ok, now I got it.

It is for reasons like that that almost all of the CI work I do is just `make <something>`. This lets me control everything from the `Makefile`, rather than worrying that something is tied too tightly into a specific command in CI.

OK, I will add a conditional `docker tag` there right away. 

At some point really soon, we are going to have to change Semaphore to do multi-arch builds with manifests. Actually, as soon as this PR is in, I will open another for that. 

Is there any chance I can convince you to change this:

```
make calico/felix
docker tag calico/felix:latest calico/felix:`git describe`
docker push  calico/felix:`git describe`
```

to something like:

```
make calico/felix
make tag
make push
```

And then I will add the `tag` and `push` targets?

Reply:
Just added another commit with:

```makefile
	ifeq ($(ARCH),amd64)
	docker tag calico/felix:latest-$(ARCH) calico/felix:latest
	endif
```

Reply:
> Is there any chance I can convince you to change this:
> 
> make calico/felix
> docker tag calico/felix:latest calico/felix:`git describe`
> docker push  calico/felix:`git describe`

Yes, I'm expecting we'll need to change it at some point.  I just don't want to have to change it more than once if I can help it so that we don't accumulate lots of conditional cruft.

Some of our build have a `make semaphore` target, which works OK.  However, felix's Semaphore build is optimised to run FV tests in lots of batches, which makes that difficult.

Reply:
> I just don't want to have to change it more than once if I can help it so that we don't accumulate lots of conditional cruft.

That makes sense. I have been working through this a little blindly since, unlike with travis or circle or Jenkins, I cannot see the CI commands we use for Semaphore.

Either way, I got the extra tagging in you requested in the right place. It looks like Semaphore either won't run or won't report until they are approved.

Once this PR is through, I will open the new one for "single make target builds all arch images", which will require those changes, but you can walk me through getting it adapted to Semaphore correctly then.

Reply:
> That makes sense. I have been working through this a little blindly since, unlike with travis or circle or Jenkins, I cannot see the CI commands we use for Semaphore

If you go to a specific run (for example, this master run: https://semaphoreci.com/calico/felix-2/branches/master/builds/921) and expand each job, you should be able to see the individual commands.